pkg/services/sso/keycloak_service_builder.go

```diff
@@ -0,0 +1,59 @@
+package sso
+
+import (
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/client/keycloak"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/client/redhatsso"
+)
+
+var _ KeycloakServiceBuilderSelector = &keycloakServiceBuilderSelector{}
+var _ KeycloakServiceBuilder = &keycloakServiceBuilder{}
+
+type KeycloakServiceBuilder interface {
+	WithRealmConfig(realmConfig *keycloak.KeycloakRealmConfig) KeycloakServiceBuilder
+	Build() KeycloakService
+}
+
+type keycloakServiceBuilder struct {
+	config      *keycloak.KeycloakConfig
+	realmConfig *keycloak.KeycloakRealmConfig
+}
+
+func (builder *keycloakServiceBuilder) Build() KeycloakService {
+	if builder.config.SelectSSOProvider == keycloak.MAS_SSO {
+		realmConfig := builder.realmConfig
+		if realmConfig == nil {
+			realmConfig = builder.config.KafkaRealm
+		}
```

Comment:
This part is duplicated. It can be de-duplicated and moved before the conditions

Reply:
Was a bug. Second should have been `RedhatSSORealm`. Fixed and changed to use the new array utils